src/Main.java

```diff
@@ -0,0 +1,63 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
```

Comment:
Metoda main() je prilis dlha a deje sa tam prilis vela veci, slo by to rozdelit aspon na nejake private metody do triedy Main alebo vytvorit nove triedy pre niektore funkcionality, napr. InputProcessor, TeamFactory, atd.

Reply:
Do bloku "try" jsem pridal podminku, ktera zajistit vytvoreni tymu jako objekty.
Neprisel jsem na to, jak bych dokazal udelat sestaveni unikatniho seznamu tymu
(neprisel jsem na to, jak vyresit porovnani objektu "String" vs. "Team")
bez pomocnyho pole "teamListUnique".
Diky tomu si ale myslim, ze se zkratil kod v metode "main" a objekty se vytvori hned na zacatku v "try"

Reply:
```
Scanner inputScanner = new Scanner(System.in);

String fileName = promptFileName(inputScanner);
List<MatchStats> matchList = new MatchAnalyzer().createMatchList(fileName);

String teamName = promptTeamName(inputScanner);
Team team = new TeamAnalyzer(matchStats).analyzeTeam(teamName);
printTeamStats(team);

inputScanner.close();
```

Trosku ta navodim: co keby metoda main() vypadala takto? Je to citelne riadok po riadku, ako ten algoritmus funguje v 'big-picture', je to kratke a jednotlive dilci casti su oddelene do metod/trid. Suhlasis s tym, ze je to cistejsi a citelnejsi? Skus pouzivat tento kod a doplnit zbyvajuce casti programu.

Reply:
Ad `teamListUnique`, pouzival by som namiesto mapu `Map<String, Team>` - budes mat jeden objekt, budes moct hladat stringy pomocou `map.keySet().contains()`, a zaroven dostanes sa aj k objektu pomocou `map.get()`